pkg/controllers/raycluster_webhook.go

```diff
@@ -68,13 +69,21 @@ var _ webhook.CustomValidator = &rayClusterWebhook{}
 func (w *rayClusterWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	rayCluster := obj.(*rayv1.RayCluster)
 
+	// add annotation to use new names
+	annotations := rayCluster.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	annotations[newNameAnnotation] = "true"
```

Comment:
While we are at it, it could be useful to generalise that annotation so that it contains the version of the operator that defaulted the Ray cluster. WDYT?